### ISSUE TITLE
Can't deploy functions to a particular project

### ISSUE THREAD

**Issue by PierBover** [firebase/firebase-functions#343]
I have a functions project that I deploy on multiple Firebase projects. It deploys fine on all projects except one.

You can see this behavior in this video: https://youtu.be/-AhA4GVLpXo

Everything is on latest versions:
* Firebase CLI: `6.0.1`
* Admin: `^6.1.0`
* Functions: `^2.1.0`

The error I get is:
```
i  deploying functions
i  functions: ensuring necessary APIs are enabled...
✔  functions: all necessary APIs are enabled
i  functions: preparing functions directory for uploading...

Error: Error occurred while parsing your function triggers.

undefined
```

You can see the complete `--debug` log here:
https://gist.github.com/PierBover/2f48cf12a18cd199105d3544612f4b8a

**Comment by PierBover** [comment]
So I created a new Firebase project and encountered the same error.

Then I tried to do `firebase serve --debug` and I got another error:
```
[2018-11-04T14:31:58.898Z] Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/firebase-tools/node_modules/@google-cloud/functions-emulator/logs'
    at Object.fs.mkdirSync (fs.js:885:18)
    at Object.assertLogsPath (/usr/local/lib/node_modules/firebase-tools/node_modules/@google-cloud/functions-emulator/src/emulator/logs.js:65:10)
    at new Controller (/usr/local/lib/node_modules/firebase-tools/node_modules/@google-cloud/functions-emulator/src/cli/controller.js:77:32)
    at FunctionsEmulator.start (/usr/local/lib/node_modules/firebase-tools/lib/functionsEmulator.js:102:23)
    at Object.start (/usr/local/lib/node_modules/firebase-tools/lib/serve/functions.js:7:33)
    at /usr/local/lib/node_modules/firebase-tools/lib/serve/index.js:18:23
    at arrayMap (/usr/local/lib/node_modules/firebase-tools/node_modules/lodash/lodash.js:639:23)
    at Function.map (/usr/local/lib/node_modules/firebase-tools/node_modules/lodash/lodash.js:9556:14)
    at _serve (/usr/local/lib/node_modules/firebase-tools/lib/serve/index.js:16:26)
    at Command._action (/usr/local/lib/node_modules/firebase-tools/lib/commands/serve.js:58:12)
```

So I had to do `sudo firebase serve` and I got this:
```
=== Serving from '/Volumes/Data/Projects/Github/media-site-cloud-functions'...

i  functions: Preparing to emulate functions.
Warning: You're using Node.js v8.11.1 but Google Cloud Functions only supports v6.11.5.
⚠  functions: Failed to load functions source code. Ensure that you have the latest SDK by running npm i --save firebase-functions inside the functions directory.
⚠  functions: Error from emulator. Error occurred while parsing your function triggers.

undefined
```

**Comment by PierBover** [comment]
Ok, I figured it out.

It was a problem on my end, but the `undefined` error didn't help at all.

The problem is that I have a dynamic env config that uses `process.env.GCLOUD_PROJECT` to get the correct config for each project. I lacked some config var for the problematic project.

**Comment by miketierce** [comment]
> Ok, I figured it out.
> 
> It was a problem on my end, but the `undefined` error didn't help at all.
> 
> The problem is that I have a dynamic env config that uses `process.env.GCLOUD_PROJECT` to get the correct config for each project. I lacked some config var for the problematic project.

Could you share more details about how you setup your process.env.GCLOD_PROJECT to accomplish deploying to firebase? I'm getting this same error attempting to deploy this project fork https://github.com/muhibbudins/nuxt-multiple

**Comment by PierBover** [comment]
I have an `environments.js` file with my different env configurations, and then something like this:

```
	getEnvironmentConfig () {
		const env = environments[process.env.GCLOUD_PROJECT];
		if (!env) throw 'No Environment!';
		return env;
	}
```

I call `getEnvironmentConfig()` from inside my cloud functions.

If I remember correctly the problem I had was that I called `getEnvironmentConfig()` before the function was executed or something like that.

**Comment by JanErikFoss** [comment]
I had the same problem, only error message I got was "undefined".

After doing a painfully long binary search for the error by commenting out half of the code each time, I tracked in down to the @woocommerce/woocommerce-rest-api constructor.

It accepts consumerKey and consumerSecret, which I was setting through functions.config().

But I had forgotten to set the config values on my development project, which is why I only saw the error on that specific project.
